agent/tcs/client/client.go

```diff
@@ -270,6 +268,17 @@ func (cs *clientServer) serviceConnectMetricsToPublishMetricRequests(requestMeta
 	tempTaskMetric := *taskMetric
 	tempTaskMetric.ServiceConnectMetricsWrapper = tempTaskMetric.ServiceConnectMetricsWrapper[:0]
 
+	// Check if taskMetric without service connect metrics exceed the message size
```

Comment:
If taskMetric could exceed max size without even adding SC metrics, should we move this block to common logic inside `metricsToPublishMetricRequets` before calling `serviceConnectMetricsToPublishMetricRequests`, so that we make this check regardless of whether we add SC metrics or not? 

And for my understanding, does this mean the issue could be happening at this moment (regardless of whether Agent supports SC or not)? Trying to see if we want to this fix should go to dev directly.

Reply:
We send sc stats every 60s and we send docker stats every 20s. The limit is either 10 tasks or 1 MB frame size limit. Without SC stats, it is exactly same as before, it should be 10 tasks per frame. That is why have this check whenever we are collecting SC stats. Does that make sense?

Reply:
Thought more about it and made the change. This will be helpful if in future we add more metrics to the docker stats metrics, we will always have that check that for 10 tasks , we are not exceeding frame size limit.